Source/Foundation/LineReflectionModule.f90

```diff
@@ -0,0 +1,226 @@
+! 
+      FUNCTION lineCoefficientTestError()  RESULT(e)
+!
+!        ---------------------------------------------------------
+!        Compute two points along a line ax+by+c = 0 and 
+!        reverse to compute a,b,c. The results should be the same 
+!        to within a constant, returning e \approx 0. 
+!        ---------------------------------------------------------
+!
+         IMPLICIT NONE  
+         REAL(KIND=RP) :: x0(3), x1(3)
+         REAL(KIND=RP) :: a, b, c
+         REAL(KIND=RP) :: aT, bT, cT
+         REAL(KIND=RP) :: e, r1, r2, r
+         
+         CALL GetReflectionTestPoints(x0,x1)
+         CALL ComputeLineCoefs(x0,x1,a,b,c)
+         CALL GetTestLineCoefficients(aT,bT,cT)
+!
+!        ----------------------------------------------------
+!        The coefficients can be scaled differently, but must
+!        all have the same scale
+!        ----------------------------------------------------
+!
+         r1 = MAX(ABS(at),ABS(bt),ABS(ct))
+         r2 = MAX(ABS(a),ABS(b),ABS(c))
+         r  = r1/r2
+         
+         e = MAX(ABS(r*a-aT), ABS(r*b-BT), ABS(r*c-cT))
+
+      END FUNCTION lineCoefficientTestError
+!
+!//////////////////////////////////////////////////////////////////////// 
+! 
+      LOGICAL FUNCTION linesAreColinear(a,b,c,at,bt,ct)
+         IMPLICIT NONE
+!
+!        ---------
+!        Arguments
+!        ---------
+!
+         REAL(KIND=RP) :: a, b, c
+         REAL(KIND=RP) :: aT, bT, cT
+!
+!        ---------------
+!        Local Variables
+!        ---------------
+!
+         REAL(KIND=RP) :: e, r1, r2, r
+ !
+!        ----------------------------------------------------
+!        The coefficients can be scaled differently, but must
+!        all have the same scale for two lines to be colinear
+!        ----------------------------------------------------
+!
+         r1 = MAX(ABS(at),ABS(bt),ABS(ct))
+         r2 = MAX(ABS(a),ABS(b),ABS(c))
+         r  = r1/r2
+         
+         e = MAX(ABS(r*a-aT), ABS(r*b-BT), ABS(r*c-cT))
+         
+         IF ( e < colinearTolerance )     THEN
+            linesAreColinear = .TRUE. 
+         ELSE 
+            linesAreColinear = .FALSE. 
+         END IF 
+        
+      END FUNCTION linesAreColinear
+!
+!//////////////////////////////////////////////////////////////////////// 
+! 
+      FUNCTION lineReflectionTestError() RESULT(e)
+!
+!        ------------------------------------------------------
+!        Reflect a point about a line and compare to the exact.
+!        ------------------------------------------------------
+!
+         IMPLICIT NONE  
+         REAL(KIND=RP) :: x0(3), x1(3), p(3), pr(3)
+         REAL(KIND=RP) :: a, b, c, e
+         REAL(KIND=RP) :: pre(3) = [0.0_RP,1.5_RP,0.0_RP]
+         
+         x0 = [0.0_RP,0.5_RP,0.0_RP]
+         x1 = [1.0_RP,1.5_RP,0.0_RP]
+         CALL ComputeLineCoefs(x0,x1,a,b,c)
+         
+         p  = [1.0,0.5,0.0]
```

Comment:
Absolutely. Can't believe I did that. Always type literals is the rule.

Reply:
Not sure what the followup question is. There can only be reflection about a single *line*. If multiple boundaries are designated as 'symmetry' lines then they must therefore be co-linear. If multiple boundaries are designated as symmetry and they are not along the same line, then an error exception is thrown saying it can't do that. Is this the case that you are thinking of?

Reply:
Ah, maybe I misunderstood. So I cannot put, say, the left boundary and the right boundary as symmetry and then reflect the mesh over each. I am only allowed to have, say, the left boundary set to symmetry and this left boundary could be composed as a chain of colinear line segments. The latter makes more sense.

Reply:
Right. I would say the first situation has two *reflection* boundaries. The second, which is what I have implemented is to have a symmetry boundary, which is a single line across which one will reflect. What you describe does not generate a symmetric mesh, I don't think. (Nope, it doesn't... I can sketch out an easy one with a graded mesh in a box.) There is nothing wrong with what you describe, but that's not implemented. Another option is to not reflect but to reproduce and shift a mesh to make a repeated pattern. The way I implemented the symmetric mesh can be extended to both of the cases (reflection and reproduction), though how to write a logical command for it would take some thought. Right now, the key is to have collinear boundaries (think of a semi-annular region with two straight lines at the ends) and reflect about that. Does all that make sense?

Reply:
Yes, that makes sense. The requested feature was that one generates half of the mesh and then reflects it to create the other half. The situation I envisioned would likely never come up because, with some "pre-processing," one could define an appropriate partial domain to mesh that then requires a single reflection. 

A natural extension of this feature would be reflecting over a `CircularArc` or `Spline`, but that should be the topic of a different PR.

Reply:
Hmm, like in complex variables with a 1/r transformation? Yeah, a different PR!